### PR TITLE
symeig supports complex backward

### LINKED ISSUE #53651: Add complex autograd support for torch.symeig
```
>>> a = torch.randn(5, 5, dtype=torch.cdouble, requires_grad=True)
>>> a = (a + a.transpose(-2, -1).conj()).div_(2)
>>> e, v = torch.symeig(a, eigenvectors=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: symeig does not support automatic differentiation for outputs with complex dtype.
```

cc @ezyang @anjali411 @dylanbespalko @mruberry @aocsa

### PR DESCRIPTION
**Pull request by lezcano**
Fixes https://github.com/pytorch/pytorch/issues/53651
I did not put much effort in improving the docs, as I will go over all these docs in future PRs
cc @anjali411

### CODE REVIEW

**Review by anjali411 — CHANGES_REQUESTED** [review]
Hi @Lezcano thanks for the PR! This PR needs tests to verify the complex autograd correctness for symeig. There are a few tests to verify symeig autograd correctness for floating point types in `test_autograd.py`. I think we can remove all those tests and add an `OpInfo` instead.

**Comment by Lezcano** [comment]
Done!

File: test/test_autograd.py [hunk 1]
```diff
@@ -2865,26 +2865,6 @@ def test_var_mean_differentiable(self):
         torch.autograd.backward(r2, grad)
         self.assertTrue(torch.allclose(input1.grad, input2.grad, rtol=0.01, atol=0.0))
 
-    @skipIfNoLapack
```
Comment (anjali411) [inline test/test_autograd.py:2868]: There are another test https://github.com/pytorch/pytorch/blob/f71a0daeb747148520ef275fa65b79cc6f5fa36b/test/test_autograd.py#L2890-L2947 that can be migrated as well.

Comment (Lezcano) [reply test/test_autograd.py:2868]: I also thought so when doing a quick search, but it turns out that that's from `torch.lobpcg`!

Comment (anjali411) [reply test/test_autograd.py:2868]: ah okay!

**Review by anjali411 — APPROVED** [review]
Thanks @Lezcano !

**Comment by anjali411** [comment]
Hey @Lezcano looks like this PR needs a rebase

**Comment by Lezcano** [comment]
@anjali411 done!

**Pull request closed without merging**